### PR TITLE
Support config server in enterprise tier

### PR DESCRIPTION
**Pull request by sophiaso**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/spring/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 Release History
 ===============
+1.20.2
```
Comment (jiec-msft) [inline src/spring/HISTORY.md:3]: Use minor version for new feature, like `1.21.0`. `1.20.1` was a mistake.

File: src/spring/azext_spring/custom.py [hunk 2]
```diff
@@ -982,18 +984,71 @@ def config_repo_update(cmd, client, resource_group, name, repo_name,
...
+
+    properties = models.ConfigServerProperties()
+    config_server_resource = models.ConfigServerResource(properties=properties)
+    logger.warning("Creating config server")
+    return client.config_servers.begin_update_put(resource_group, service, config_server_resource)
+
+def config_delete(cmd, client, service, resource_group):
+    return client.config_servers.begin_delete(resource_group, service)
+
+def config_bind(cmd, client, service, resource_group, app):
```
Comment (jiec-msft) [inline src/spring/azext_spring/custom.py:1015]: I didn't find the code for the `app` param in the `_param.py` file. Is it not needed or I missed it.

File: src/spring/azext_spring/commands.py [hunk 3]
```diff
@@ -158,15 +158,19 @@ def load_command_table(self, _):
         g.custom_command('enable', 'eureka_enable')
         g.custom_command('disable', 'eureka_disable')
 
-    with self.command_group('spring config-server', client_factory=cf_config_servers,
+    with self.command_group('spring config-server',
                             exception_handler=handle_asc_exception) as g:
         g.custom_command('set', 'config_set', supports_no_wait=True)
-        g.custom_command('clear', 'config_delete')
+        g.custom_command('clear', 'config_clear')
```
[the remark this answers is not in the document]
Comment (sophiaso) [reply src/spring/azext_spring/commands.py]: This only changes the method name, which should not affect customer's experience, so it's not breaking change.

**Review by yuwzho — APPROVED** [review]

File: src/spring/azext_spring/commands.py [hunk 4]
```diff
@@ -158,19 +158,23 @@ def load_command_table(self, _):
         g.custom_command('enable', 'eureka_enable')
         g.custom_command('disable', 'eureka_disable')
 
-    with self.command_group('spring config-server', client_factory=cf_config_servers,
+    with self.command_group('spring config-server',
                             exception_handler=handle_asc_exception) as g:
         g.custom_command('set', 'config_set', supports_no_wait=True)
-        g.custom_command('clear', 'config_delete')
+        g.custom_command('clear', 'config_clear', confirmation=True)
```
Comment (yanzhudd) [inline src/spring/azext_spring/commands.py:164]: It is better to specify this type of change to the changelog

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**